4-storage/kinesis-elasticsearch-sink/src/main/scala/com.snowplowanalytics.snowplow.storage.kinesis/elasticsearch/clients/ElasticsearchSenderHTTP.scala

```diff
@@ -0,0 +1,234 @@
+
+// Jest
+import io.searchbox.client.{
+  JestClient,
+  JestClientFactory
+}
+import io.searchbox.core._
+import io.searchbox.client.config.HttpClientConfig
+import io.searchbox.cluster.Health
+
+// Java
+import java.util.concurrent.TimeUnit;
+
+// Joda-Time
+import org.joda.time.{DateTime, DateTimeZone}
+import org.joda.time.format.DateTimeFormat
+
+// Scala
+import scala.collection.JavaConversions._
+import scala.annotation.tailrec
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// json4s
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+import org.json4s.JsonDSL._
+
+// Logging
+import org.apache.commons.logging.{
+  Log,
+  LogFactory
+}
+
+// Tracker
+import com.snowplowanalytics.snowplow.scalatracker.Tracker
+
+// This project
+import sinks._
+
+/**
+ * Sends Elasticsearch documents via the HTTP Jest Client.
+ * Batches are bulk sent to the end point.
+ */
+class ElasticsearchSenderHTTP(
+  configuration: KinesisConnectorConfiguration,
+  tracker: Option[Tracker] = None,
+  maxConnectionWaitTimeMs: Long = 60000
+) extends ElasticsearchSender {
+
+  private val Log = LogFactory.getLog(getClass)
+
+  // An ISO valid timestamp formatter
+  private val TstampFormat = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").withZone(DateTimeZone.UTC)
+
+  /**
+   * Prepare the elasticsearch client
+   */
+  private val factory: JestClientFactory = new JestClientFactory()
+  factory.setHttpClientConfig(new HttpClientConfig
+    .Builder("http://" + configuration.ELASTICSEARCH_ENDPOINT + ":" + configuration.ELASTICSEARCH_PORT)
+    .multiThreaded(true)
+    .discoveryEnabled(false)
+    .maxConnectionIdleTime(30L, TimeUnit.SECONDS)
+    .connTimeout(5000)
+    .readTimeout(5000)
+    .build()
+  )
+  private val elasticsearchClient: JestClient = factory.getObject()
+
+  /**
+   * The Elasticsearch endpoint.
+   */
+  private val elasticsearchEndpoint = configuration.ELASTICSEARCH_ENDPOINT
+
+  /**
+   * The Elasticsearch port.
+   */
+  private val elasticsearchPort = configuration.ELASTICSEARCH_PORT
+
+  /**
+   * The amount of time to wait in between unsuccessful index requests (in milliseconds).
+   * 10 seconds = 10 * 1000 = 10000
+   */
+  private val BackoffPeriod = 10000
+       
+  Log.info("ElasticsearchSender using elasticsearch endpoint " + elasticsearchEndpoint + ":" + elasticsearchPort)
+
+  /**
+   * Emits good records to Elasticsearch and bad records to Kinesis.
+   * All valid records in the buffer get sent to Elasticsearch in a bulk request.
+   * All invalid requests and all requests which failed transformation get sent to Kinesis.
+   *
+   * @param records List of records to send to Elasticsearch
+   * @return List of inputs which Elasticsearch rejected
+   */
+  def sendToElasticsearch(records: List[EmitterInput]): List[EmitterInput] = {
+
+    val actions: List[io.searchbox.core.Index] = for {
+      (_, Success(record)) <- records
+    } yield {
+      new io.searchbox.core.Index.Builder(record.getSource)
```

Comment:
Any reason why there are no `io.searchbox....` imports in the file instead of this long-form?